### ISSUE TITLE
Writing to the same file from multiple processes does not function when archiving with concurrentWrites="true" and keepFileOpen="true"

### ISSUE THREAD

**Issue by thihasoehtet** [NLog/NLog#1513]
**Type**: Bug
**NLog version**: 4.3.5
**Platform**: .Net 4.5
**Current NLog config**

``` xml
  <nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <targets>
      <target name="file"
              xsi:type="File"
              layout="${longdate} (${threadid}) ${level} [${logger}] ${message}"
              fileName="C:/NLogTest/logfile.log"
              archiveFileName="C:/NLogTest/{##}-logfile.log"
              archiveAboveSize="100000"
              archiveEvery="Minute"
              archiveNumbering="DateAndSequence"
              archiveDateFormat="yyyyMMdd"
              maxArchiveFiles="30"
              concurrentWrites="true"
              keepFileOpen="true"
              encoding="iso-8859-2" />
      <target name="Console" type="console" />
    </targets>
    <rules>
      <logger name="*" minlevel="Debug" writeTo="console,file" />
    </rules>
  </nlog>
```

**Test Program**

``` cs
using System;
using System.Diagnostics;
using System.Reflection;
using System.Threading;
using NLog;

namespace ConcurrentFileTargetTest
{
  class Program
  {
    private static readonly ILogger Logger = LogManager.GetCurrentClassLogger();
    static void Main(string[] args)
    {
      string processName;
      if (args.Length == 0)
      {
        processName = "0";
        for (int i = 1; i < 5; ++ i)
        {
          Process.Start(Assembly.GetEntryAssembly().Location, i.ToString());
        }
      }
      else
      {
        processName = args[0];
      }

      bool running = true;
      Console.CancelKeyPress += (sender, eventArgs) => running = false; // shutdown on Control-C
      while (running)
      {
        Logger.Debug("{0} - Happily Running Still", processName);
        Thread.Sleep(1000);
      }
    }
  }
}
```

**What is the current result?**
This is similar to an old issue [#1029].
After archiving, one process is writing to new log file (logfile.log) but other processes are writing to a separate archived log files, likes **process1** is writing to the **logfile.log** but **process2** is writing to **20160620.01-logfile.log** and **process3** is writing to **20160620.02-logfile.log**.
It is able to reproduce with the same test application of the old issue[#1029] and seems
it is still happening like what **_bhaeussermann_** mentioned in the old issue,

> So with the current release of NLog, the following happens: 
> 1) The first process to log after changing the date, archives the current log file, and its appender is invalidated. A new appender is created for the new log file. 
> 2) This causes the last-write-time of the archived file to be updated.
> 3) The second process still has a handle to the now-archived file. It reads the updated last-write-time and concludes that the file must not be archived. It continues logging to that archived file.

**What is the expected result?**
After archiving, all processes should be writing to a single new file (logfile.log)

**Is there a version in which it did worked?** 
We are just upgrading from **2.1.0** to the latest version **4.3.5** and it was working in **2.1.0**

**Comment by 304NotModified** [comment]
Thabks for opening a new issue, it's easier to follow up. 

@bhaeussermann any idea on this?

**Comment by 304NotModified** [comment]
Fixed by #1771
